src/cmds/scripts/post_install.py

```diff
@@ -0,0 +1,463 @@
+	buffer="set PBS_CONF_FILE=" + pbs_path + "pbs.conf\r\n"
+	file.write(buffer)
+	# initialize database cluster
+	buffer=pbs_pathquote + "exec\\pgsql\\bin\\initdb.exe\"" + " -D " + DBpath + " -E SQL_ASCII --locale=C\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# start database
+	buffer="call " + pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " start\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# create db
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\createdb.exe\" -p 15007 pbs_datastore\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# install schema
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\psql.exe\" -p 15007 -d pbs_datastore -f "+pbs_pathquote+"exec\\etc\\pbs_db_schema.sql\"\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# create superuser 
+	buffer=pbs_pathquote+"exec\\pgsql\\bin\\psql.exe\" -p 15007 -d pbs_datastore -c \"create user \\\"" + username + "\\\" SUPERUSER LOGIN\"\r\n"
+	file.write(buffer)
+	# run Pbs_ds_password
+	buffer=pbs_pathquote+"exec\\sbin\\pbs_ds_password.exe\" -r\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# stop database
+	buffer="call "+ pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " stop\r\n"
+	file.write(buffer)
+	file.write(ErrChk)
+	# error handling
+	file.write("exit /b 0\r\n")
+	file.write(":ERR\r\n")
+	# stop database
+	buffer="call "+ pbs_pathquote + "exec\\sbin\\pbs_dataservice\"" + " stop\r\n"
+	file.write(buffer)
+	file.write("exit /b 1\r\n")
+	file.close()
+	print "Executing createdbfile as install user"
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen([createdbfile], stdout=fd, stderr=fd)
+			theproc.communicate()
+			#print theproc.returncode
+			if theproc.returncode > 0:
+				print "Failed to initilize database"
+				sys.exit(3)
+			else:
+				print "Database intializaation success"
+				
+	UpdateDSconfig()
+	print "InstallDataService completed... "
+	
+	
+def CreateServerdbFile():
+	#os.system("net stop pbs_server")
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen(["sc", "stop", "pbs_server"], stdout=fd, stderr=fd)
+			theproc.communicate()
+			
+	# Creating PBS Server database...
+	ServerPath=pbs_path+"exec\\sbin\\pbs_server.exe"
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen([ServerPath, "-C"], stdout=fd, stderr=fd)
+			theproc.communicate()
+			#print theproc.returncode
+			if theproc.returncode > 0:
+				print "Failed to create server database..."
+				sys.exit(7)
+			else:
+				print "PBS Server database created..."
+	with open(os.devnull, 'w') as fd:
+			theproc= subprocess.Popen(["sc", "start", "pbs_server"], stdout=fd, stderr=fd)
+			theproc.communicate()
+			time.sleep(60)
+	# create server default file
+	Serverdfltfile=pbs_path+"home\\spool\\createsvrdefaults.bat"
+	QmgrPath=pbs_path+"exec\\bin\\qmgr.exe"
+	if ( os.path.exists(QmgrPath) ):
+		pass
+	else:
+		print "Qmgr not available"
+		sys.exit(8)
+	quotedQmgrPath="\""+QmgrPath+ "\""
+	ErrChk = "IF %ERRORLEVEL% NEQ 0 GOTO ERR\r\n"
+	file = open(Serverdfltfile,"w")
```

Comment:
You should use a with statement when opening a file to ensure it gets closed, even if something goes wrong.